src/api/spec/features/beta/webui/maintenance_workflow_spec.rb

```diff
@@ -0,0 +1,126 @@
+           title: 'official maintenance space',
+           target_project: [update_project, another_update_project],
+           maintainer: maintenance_coord_user)
+  end
+  let(:user) { create(:confirmed_user, :with_home, login: 'tom') }
+  let(:maintenance_coord_user) { create(:confirmed_user, :with_home, login: 'maintenance_coord') }
+  let(:project) { create(:project_with_repository, name: 'openSUSE:11.4') }
+  let(:another_project) { create(:project_with_repository, name: 'openSUSE:11.5') }
+  let!(:maintenance_package) { create(:package_with_file, project: project, name: 'cacti') }
+  let!(:another_package) { create(:package_with_file, project: another_project, name: 'cacti') }
+  let(:update_project) { create(:update_project, maintained_project: project, name: "#{project}:Update") }
+  let(:another_update_project) { create(:update_project, maintained_project: another_project, name: "#{another_project}:Update") }
+  let(:bs_request) { BsRequest.last }
+
+  before do
+    User.session = admin_user
+    create(:maintenance_project_attrib, project: maintenance_project)
+  end
+
+  it 'maintenance workflow without patchinfo' do
+    # # Step 1: The user branches a package
+    # ####################################
+    login(user)
+
+    # visit package_show_path(project: update_project, package: maintenance_package)
+    # desktop? ? click_link('Branch Package') : click_menu_link('Actions', 'Branch Package')
+    # expect(page).to have_text('Source')
+
+    # click_button('Branch')
+
+    # expect(page).to have_text('Successfully branched package')
+
+    target_packages = []
+    user.run_as do
+      target_packages << BranchPackage.new(project: project.name, package: maintenance_package.name, target_project: 'home:tom:branches:Update',
+                                           target_package: 'cacti.openSUSE:11.4').branch
+    end
+    user.run_as do
+      target_packages << BranchPackage.new(project: another_project.name, package: another_package.name, target_project: 'home:tom:branches:Update',
+                                           target_package: 'cacti.openSUSE:11.5').branch
+    end
+    update_project_branch = Project.last
+    target_packages.each do |package|
+      Backend::Connection.put("/source/#{package[:data][:targetproject]}/#{package[:data][:targetpackage]}/DUMMY_FILE", 'dummy')
+    end
+
+    # Create the patchinfo
+    visit project_show_path(project: Project.last.name)
+    desktop? ? click_link('Create Patchinfo') : click_menu_link('Actions', 'Create Patchinfo')
+    fill_in('patchinfo_summary', with: 'ProjectWithRepo_package is much better than the old one')
+    fill_in('patchinfo_description', with: 'Fixes nothing, Fixes nothing, Fixes nothing, Fixes nothing, Fixes nothing, Fixes nothing')
+    check('patchinfo_block')
+    fill_in('patchinfo_block_reason', with: 'locked!')
+    click_button('Save')
+    expect(page).to have_css('#flash', text: 'Successfully edited patchinfo')
```

Comment:
This is the setup you should do via factories.

- https://www.betterspecs.org/#let
- https://www.betterspecs.org/#factories